rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs

```diff
@@ -131,15 +132,15 @@ where
     // Smithy output must convert into a HTTP response
     Op::Output: IntoResponse<P>,
     // Smithy error must convert into a HTTP response
-    OpError: IntoResponse<P>,
+    Op::Error: IntoResponse<P>,
 
     // Must be able to convert extensions
     Exts: FromParts<P>,
 
     // The signature of the inner service is correct
-    S: Service<(Op::Input, Exts), Response = Op::Output, Error = OperationError<OpError, PollError>>,
+    S: Service<(Op::Input, Exts), Response = Op::Output, Error = Op::Error> + Clone,
 {
-    type Output = Result<http::Response<crate::body::BoxBody>, PollError>;
+    type Output = Result<http::Response<crate::body::BoxBody>, Infallible>;
```

Comment:
Can't this just be:
```suggestion
    type Output = http::Response<crate::body::BoxBody>;
```

Reply:
It can be but then we'd have to `map` in the `Service::call` method which would leak into the `type Future = `, which we don't want.